jmx/plugins/org.jboss.tools.jmx.core/src/org/jboss/tools/jmx/core/tree/ObjectNameNode.java

```diff
@@ -62,17 +77,24 @@ public void run(MBeanServerConnection mbsc) throws Exception {
 		    	});
         	}
     	} catch( Exception e ) {
-    		// TODO FIX THIS
+    		JMXActivator.getDefault().getLog().log(new Status(IStatus.ERROR, JMXActivator.PLUGIN_ID, "Error loading object name details", e));
+    		loadError = e;
     	}
     	wrapper = array[0];
     }
 
-
+    public boolean hasLoadError() {
+    	return loadError != null;
+    }
+    
     public ObjectName getObjectName() {
         return on;
     }
 
-    public MBeanInfoWrapper getMbeanInfoWrapper() {
+    public synchronized MBeanInfoWrapper getMbeanInfoWrapper() {
+    	if( wrapper == null ) {
```

Comment:
how often the loadInfo is called? In case of loadError will it be recalled a lot of times? only on refresh?